src/file_pipe_log/pipe.rs

```diff
@@ -782,19 +483,20 @@ mod tests {
         }
         pipe_log.rotate().unwrap();
         let (first, last) = pipe_log.file_span();
+        // By `purge`, all stale files will be automatically RENAME to reserved files.
         assert_eq!(pipe_log.purge_to(last).unwrap() as u64, last - first);
         // Try to read stale file.
-        for (i, handle) in handles.into_iter().enumerate() {
+        for (_, handle) in handles.into_iter().enumerate() {
             assert!(pipe_log.read_bytes(handle).is_err());
             // Bypass pipe log
             let mut reader = build_file_reader(
                 fs.as_ref(),
-                Arc::new(fs.open(handle.id.build_file_path(path)).unwrap()),
+                Arc::new(fs.open(handle.id.build_stale_file_path(path)).unwrap()),
             )
             .unwrap();
-            assert_eq!(reader.read(handle).unwrap(), content(i));
+            assert!(reader.read(handle).unwrap().is_empty());
             // Delete file so that it cannot be reused.
```

Comment:
The original intention of this test is broken. You should either remove the comment or try some other way to test it.